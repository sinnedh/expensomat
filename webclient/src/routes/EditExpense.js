import React from 'react'
import { connect } from 'react-redux'
import EditableInput from '../components/EditableInput'
import { updateExpense } from '../actions'

class EditExpense extends React.Component {
  render() {
    return (
      <div>
        <h1>Edit expense:</h1>
        <div>
          <label>Description:</label>{' '}
          <EditableInput
            value={this.props.description}
            onClickSave={value => this.props.onUpdateDescription(value)}
            />
        </div>
        <div>
          <label>Amount:</label>{' '}
          <EditableInput
            value={this.props.amount}
            onClickSave={value => this.props.onUpdateAmount(value)}
            />
        </div>
        <div>
          <label>Paid at:</label>{' '}
          <EditableInput
            value={this.props.paid_at}
            onClickSave={value => this.props.onUpdatePaidAt(value)}
            />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  const expenseId = parseInt(ownProps.match.params.id, 10)
  const expense = state.expenses.items.find(e => e.id === expenseId)

  return {
    token: state.application.token,
    description: expense.description,
    amount: expense.amount,
    paid_at: expense.paid_at,
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  const expenseId = parseInt(ownProps.match.params.id, 10)

  return {
    onUpdateDescription: (description) => {
      dispatch(updateExpense(ownProps.token, expenseId, {description}))
    },
    onUpdateAmount: (amount) => {
      dispatch(updateExpense(ownProps.token, expenseId, {amount}))
    },
    onUpdatePaidAt: (paid_at) => {
      dispatch(updateExpense(ownProps.token, expenseId, {paid_at}))
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense)
