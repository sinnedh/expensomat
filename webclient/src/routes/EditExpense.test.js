import React from 'react'
import { mount, shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import EditExpense from './EditExpense'

describe('EditExpense', () => {
  it('renders properly', () => {
    const initialState = {
      application: { token: 'ABCD1234' },
      expenses: {
        items: [
          {id: 1, description: 'Expense 1', amount: 999, paid_at: '2017-11-11'},
          {id: 2, description: 'Expense 2', amount: 1599, paid_at: '2016-12-08'}
        ]
      }
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<EditExpense match={{params: {id: 1}}} store={store} />)
    expect(wrapper.dive()).toMatchSnapshot()
  })
})
