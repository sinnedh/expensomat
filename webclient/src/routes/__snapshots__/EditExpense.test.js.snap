// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExpense renders properly 1`] = `
<div>
  <h1>
    Edit expense:
  </h1>
  <div>
    <label>
      Description:
    </label>
     
    <EditableInput
      onClickSave={[Function]}
      value="Expense 1"
    />
  </div>
  <div>
    <label>
      Amount:
    </label>
     
    <EditableInput
      onClickSave={[Function]}
      value={999}
    />
  </div>
  <div>
    <label>
      Paid at:
    </label>
     
    <EditableInput
      onClickSave={[Function]}
      value="2017-11-11"
    />
  </div>
</div>
`;
